src/openssh/patch/0004-ssh-crash-fix.patch

```diff
@@ -0,0 +1,107 @@
+From 745b435a3bf705987ed546e32bd9fe0c6e5378f3 Mon Sep 17 00:00:00 2001
+From: Asha Behera <asha.behera@broadcom.com>
+Date: Mon, 24 Feb 2025 23:19:10 -0700
+Subject: [PATCH] [PATCH] ssh crash fix
+
+---
+ auth.c    | 9 +++++++--
+ auth.h    | 2 +-
+ packet.c  | 2 +-
+ session.c | 5 ++++-
+ sshd.c    | 6 ++++--
+ 5 files changed, 17 insertions(+), 7 deletions(-)
+
+diff --git a/auth.c b/auth.c
+index 73a85fd..4b4a7b2 100644
+--- a/auth.c
++++ b/auth.c
+@@ -1191,12 +1191,17 @@ auth_authorise_keyopts(struct ssh *ssh, struct passwd *pw,
+ }
+ 
+ /* Export remote IP address and port for authorization. */
+-void
++int
+ export_remote_info(struct ssh *ssh)
+ {
++        if (ssh == NULL)
++        {
++          return 1;
++        }
+ 	const char *remote_ip = ssh_remote_ipaddr(ssh);
+ 	const int remote_port = ssh_remote_port(ssh);
+ 	const char remote_addr_port[32 + INET6_ADDRSTRLEN];
+ 	snprintf(remote_addr_port, sizeof(remote_addr_port), "%s %d", remote_ip, remote_port);
+ 	setenv("SSH_CLIENT_IPADDR_PORT", remote_addr_port, 1);
+-}
+\ No newline at end of file
++        return 0; 
++}
+diff --git a/auth.h b/auth.h
+index a83d31d..7186177 100644
+--- a/auth.h
++++ b/auth.h
+@@ -234,7 +234,7 @@ pid_t	subprocess(const char *, struct passwd *,
+ int	 sys_auth_passwd(struct ssh *, const char *);
+ 
+ /* Export remote IP address and port for authorization. */
+-void	 export_remote_info(struct ssh *);
++int	 export_remote_info(struct ssh *);
+ 
+ #if defined(KRB5) && !defined(HEIMDAL)
+ krb5_error_code ssh_krb5_cc_gen(krb5_context, krb5_ccache *);
+diff --git a/packet.c b/packet.c
+index 00e3180..f53ce95 100644
+--- a/packet.c
++++ b/packet.c
+@@ -529,7 +529,7 @@ ssh_remote_ipaddr(struct ssh *ssh)
+ 	int sock;
+ 
+ 	/* Check whether we have cached the ipaddr. */
+-	if (ssh->remote_ipaddr == NULL) {
++	if (ssh != NULL && ssh->remote_ipaddr == NULL) {
```

Comment:
This part of code belong to sshd, suggest don't change it's behavior.

Reply:
@liuh-80 : Do you see any issue with the check ? It is good to have check of ssh NULL before accessing its contents remote_ipaddr.

Reply:
This part of code never crashes in official openssh, so I think ssh will never be NULL here, if we make the change, the risk is this patch may break with future version.

I also agree with you that the new code does not have issue. so just a suggestion.

Reply:
removing this change for minimal change